### PR TITLE
Update project to Golang 1.18

### PR DESCRIPTION
**Pull request by ervitis**
Hello! I may use this project and we are using golang 1.18 version, so I updated it with some features:

- Use go.mod
- Add build constraints
- Add gitignore file

Thank you for this great work!

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/ervitis/filecache
```
Comment (kisom) [inline go.mod]: This isn't right for this repo; it would work for your own fork.

Comment (ervitis) [reply go.mod]: @kisom sorry! I will update it

**Review by kisom — CHANGES_REQUESTED** [review]
One little tweak; can you change the gomod package name? Other than that, LGTM.

**Comment by ervitis** [comment]
@kisom sorry, I didn't understand, do you mean this name from the `go.mod` file?
Screenshot 2022-08-03 at 13 38 54

**Review by kisom — APPROVED** [review]

**Comment by kisom** [comment]
Thanks @ervitis !

**Pull request merged**